File: src/test/java/io/mfedirko/common/infra/security/OAuthLoginControllerTest.java
package io.mfedirko.common.infra.security;

import org.hamcrest.Matchers;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.servlet.MockMvc;

import java.util.Set;

import static org.hamcrest.Matchers.*;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

@SpringBootTest
@AutoConfigureMockMvc
class OAuthLoginControllerTest {
    @Autowired
    MockMvc mockMvc;

    @Test
    void loginPageListsProvidersDynamically() throws Exception {
        mockMvc.perform(get("/oauth-login"))
                .andExpect(status().isOk())
                .andExpect(content().string(allOf(
                                containsStringIgnoringCase("login with github"),
                                containsStringIgnoringCase("login with google"),
                                not(containsStringIgnoringCase("secret")))
                        ));
    }
}

File: src/test/kotlin/io/mfedirko/common/infra/security/OAuthLoginControllerTest.kt
package io.mfedirko.common.infra.security

import org.hamcrest.Matchers
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers

@WebMvcTest(controllers = [OAuthLoginController::class])
@Import(WebSecurityConfig::class)
@AutoConfigureMockMvc
internal class OAuthLoginControllerTest {
    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    @Throws(Exception::class)
    fun loginPageListsProvidersDynamically() {
        mockMvc.perform(MockMvcRequestBuilders.get("/oauth-login"))
            .andExpect(MockMvcResultMatchers.status().isOk)
            .andExpect(
                MockMvcResultMatchers.content().string(
                    Matchers.allOf(
                        Matchers.containsStringIgnoringCase("login with github"),
                        Matchers.containsStringIgnoringCase("login with google"),
                        Matchers.not(Matchers.containsStringIgnoringCase("secret"))
                    )
                )
            )
    }
}